diff -rup qemu-kvm-1.0.1/scripts/tracetool z/scripts/tracetool
--- qemu-kvm-1.0.1/scripts/tracetool	2012-04-16 22:15:17.000000000 -0400
+++ z/scripts/tracetool	2012-07-29 21:10:51.326868987 -0400
@@ -500,6 +500,15 @@ EOF
         if [ "$arg" = "limit" ]; then
           arg="_limit"
         fi
+        if [ "$arg" = "in" ]; then
+          arg="_in"
+        fi
+        if [ "$arg" = "next" ]; then
+          arg="_next"
+        fi
+        if [ "$arg" = "self" ]; then
+          arg="_self"
+        fi
         cat <<EOF
   $arg = \$arg$i;
 EOF
