diff -rup qemu-kvm-0.15.1/hw/vga.c frob/hw/vga.c
--- qemu-kvm-0.15.1/hw/vga.c	2011-10-19 09:54:48.000000000 -0400
+++ frob/hw/vga.c	2012-07-29 21:00:27.389677588 -0400
@@ -149,7 +149,7 @@ static uint16_t expand2[256];
 static uint8_t expand4to8[16];
 
 static void vga_screen_dump(void *opaque, const char *filename);
-static char *screen_dump_filename;
+static const char *screen_dump_filename;
 static DisplayChangeListener *screen_dump_dcl;
 
 static void vga_dumb_update_retrace_info(VGACommonState *s)
@@ -2330,7 +2330,6 @@ static void vga_save_dpy_update(DisplayS
 {
     if (screen_dump_filename) {
         ppm_save(screen_dump_filename, ds->surface);
-        screen_dump_filename = NULL;
     }
 }
 
@@ -2408,8 +2407,8 @@ static void vga_screen_dump(void *opaque
     if (!screen_dump_dcl)
         screen_dump_dcl = vga_screen_dump_init(s->ds);
 
-    screen_dump_filename = (char *)filename;
+    screen_dump_filename = filename;
     vga_invalidate_display(s);
     vga_hw_update();
+    screen_dump_filename = NULL;
 }
-
